hadoop-ozone/dist/src/main/smoketest/upgrade/snapshot.robot

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+*** Settings ***
+Documentation       Smoketest ozone cluster snapshot feature
+Library             OperatingSystem
+Library             BuiltIn
+Resource            ../commonlib.robot
+Test Timeout        5 minutes
+
+*** Variables ***
+
+
+*** Test Cases ***
+Create snapshot
+    [Tags]     snapshot-enabled
+    ${output} =         Execute                ozone sh volume create snapvolume-1
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute                ozone sh bucket create /snapvolume-1/snapbucket-1
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute                ozone sh snapshot create /snapvolume-1/snapbucket-1 snapshot1
+                        Should not contain     ${output}       Failed
+                        Execute and checkrc    echo "key created using Ozone Shell" > /tmp/sourcekey    0
+                        Execute                ozone sh key put /snapvolume-1/snapbucket-1/key1 /tmp/sourcekey
+                        Execute                ozone sh snapshot create /snapvolume-1/snapbucket-1 snapshot2
+
+Attempt to create snapshot when snapshot feature is disabled
+    [Tags]     snapshot-disabled
+    ${output} =         Execute And Ignore Error    ozone sh volume create snapvolume-2     
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute And Ignore Error    ozone sh bucket create /snapvolume-2/snapbucket-1     
+                        Should not contain     ${output}       Failed
+    ${rc}               ${output} =          Run And Return Rc And Output       ozone sh snapshot create /snapvolume-2/snapbucket-1 snapshot1
+                        Should be True        ${rc} != 0
```

Comment:
For checking the return code, `Execute and checkrc` from `lib.resource` is more idiomatic. This is also what `upgrade-ec-check.robot` uses for checks like this:
```robot
${result} =     Execute and checkrc     ozone sh key put -r rs-3-2-1024k -t EC /ectest/testpropchange/core-site.xml /etc/hadoop/core-site.xml     255
Should contain  ${result}   NOT_SUPPORTED_OPERATION
```

Reply:
yup sure! updated to check return error code 255.